polaris.shopify.com/src/components/GlobalSearch/GlobalSearch.tsx

```diff
@@ -103,6 +103,12 @@ function GlobalSearch({}: Props) {
     };
   }, [setIsOpen, router.events]);
 
+  useEffect(() => {
+    if (!isOpen) {
+      setSearchTerm("");
+    }
+  }, [isOpen]);
```

Comment:
We probably don't want to add a useEffect for this functionality. It would be best if when the user clicked the background it automatically closed the modal and reset the search state.

If you wanted to do this, I would create a new function to replace what happens on line #L206 in the `onClick` to be a new function called `handleClose` that closes the modal and runs `setSearchTerm("")`.


Let me know if you need any help!

Reply:
I think useEffect is appropriate here since HeadlessUI handles a lot of the opening/closing. If we create a custom handler we have to make sure that we use it for:

- Any close buttons
- Clicking the backdrop
- Pressing escape

Any any other scenario that might trigger a close. IMO, using a useEffect here seems like a good way to simply extend "whatever decided to close the modal".

Reply:
The reason why I opted to using useEffect as opposed to creating a handleClose function like you recommended is because the `onClick` and even the `onClose` prop for Headless UI only trigger when the model is closed by using the close button shown in the mobile view or clicking the background. But when clicking a search item it seems that the handler is not used.

Using useEffect was also an easier way to ensure that I was taking care of all the close events at once.